src/main/java/reposense/system/DashboardServer.java

```diff
@@ -0,0 +1,58 @@
+package reposense.system;
+
+import java.awt.Desktop;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import net.freeutils.httpserver.HTTPServer;
+import reposense.RepoSense;
+
+/**
+ * Represents the RepoSense dashboard server
+ */
+public class DashboardServer {
+
+    private static final String LOCAL_HOST_URL = "http://localhost:%s/";
+
+    private static final Logger logger = LogsManager.getLogger(RepoSense.class);
+
+    /**
+     * Starts a server at {@code port} and loads the {@code requestPath} from the local disk.
+     */
+    public static void startServer(int port, Path requestPath) {
+        logger.info("Starting a server at port " + port + "...");
+        HTTPServer server = new HTTPServer(port);
+
+        // default virtual host
+        HTTPServer.VirtualHost host = server.getVirtualHost(null);
+
+        try {
+            // a handler to process the request and give the corresponding response
+            host.addContext(File.separator, new HTTPServer.FileContextHandler(requestPath.toFile()));
```

Comment:
Worked on Windows by replacing the `File.separator` with `/` forward slash instead.